imsim/stamp.py

```diff
@@ -427,21 +429,22 @@ def buildPSF(self, config, base, gsparams, logger):
             # For FFT-rendered objects, the telescope vignetting isn't
             # emergent as it is for the ray-traced objects, so use the
             # empirical vignetting function, if it's available, to
-            # recompute the realized flux.
+            # scale the realized flux.
             if self.vignetting is not None:
                 pix_to_fp = self.det.getTransform(cameraGeom.PIXELS,
                                                   cameraGeom.FOCAL_PLANE)
-                flux = self.gal.flux*self.vignetting.at_sky_coord(
+                vignetted_flux = self.realized_flux*self.vignetting.at_sky_coord(
                     base['sky_pos'], self.image.wcs, pix_to_fp)
-                self.realized_flux = galsim.PoissonDeviate(self.rng, mean=flux)()
+                self.realized_flux = round(vignetted_flux)
 
-            logger.warning('Yes. Use FFT for this object.  max_sb = %.0f > %.0f',
```

Comment:
Sorry... I thought I added a comment here before but somehow,  I don't see it now so maybe I didn't save it correctly.

Can we change these from WARNING to level 2?